### PR TITLE
BaSP-TG-32: List/Delete Task Page

### PR DESCRIPTION
**Pull request by LucianoFranchino**
Funcionality complete. 
¡No css finish yet!
Contains: 
-Index
-FormTask: show the list of task
-ListTask: Contain the button to eliminate task
_Modal: the modal to eliminate task

### CODE REVIEW

File: src/Components/Tasks/FormTask.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          ) : (
+            list.map((task) => {
+              return <ListTask key={task._id} listTask={task} handleDelete={handleDelete} />;
+            })
+          )}
+        </tbody>
+      </table>
+    </div>
+  );
+};
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/FormTask.jsx:25]: Resolved

File: src/Components/Tasks/FormTask.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <div>
+      <table>
+        <thead>
+          <tr>
+            <th>Task</th>
+          </tr>
+        </thead>
+        <tbody>
+          {list === undefined || list.length === 0 ? (
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/FormTask.jsx:14]: Resolved

File: src/Components/Tasks/Modal/modalIndex.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <div className={styles.container}>
+      <div className={styles.modal}>
+        <h3>Delete Task</h3>
+        <button onClick={props.closeModal}>Close</button>
+        <button
+          onClick={() => {
+            props.showModal;
+            props.handleDelete(props.listTask._id);
+            props.closeModal;
+            setTimeout(function () {
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/Modal/modalIndex.jsx:18]: Resolved

File: src/Components/Tasks/listTask.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import Modal from './Modal/modalIndex';
+
+const ListTask = ({ listTask, handleDelete }) => {
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/listTask.jsx:4]: Resolved

File: src/Components/Tasks/listTask.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    setShowModal(true);
+  };
+
+  return (
+    <tr>
+      <td>{listTask.description}</td>
+      <td>
+        <button onClick={openModal}>X</button>
+        <Modal
+          listTask={listTask}
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/listTask.jsx:21]: Resolved

File: src/Components/Tasks/FormTask.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import ListTask from './listTask';
+
+const List = ({ list, handleDelete }) => {
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/FormTask.jsx:4]: Resolved

File: src/Components/Tasks/FormTask.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/FormTask.jsx:25]: Resolved

File: src/Components/Tasks/FormTask.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/FormTask.jsx:14]: Resolved

File: src/Components/Tasks/Modal/modalIndex.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/Modal/modalIndex.jsx:18]: Resolved

File: src/Components/Tasks/listTask.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/listTask.jsx:4]: Resolved

File: src/Components/Tasks/listTask.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  const openModal = () => {
+    setShowModal(true);
+  };
+
+  return (
+    <tr>
+      <td>{listTask.description}</td>
+      <td>
+        <button onClick={openModal}>X</button>
+        <Modal
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/listTask.jsx:20]: Resolved

File: src/Components/Tasks/listTask.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/listTask.jsx:21]: Resolved

File: src/Components/Tasks/FormTask.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/Components/Tasks/FormTask.jsx:4]: Resolved

File: src/Components/Tasks/Modal/index.jsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import styles from './modal.module.css';
+
+function Modal(props) {
+  if (!props.showModal) {
+    return null;
+  }
+  console.log(props.listTask);
```
Comment (lmalvasia) [inline src/Components/Tasks/Modal/index.jsx:7]: Remove this console.log

File: src/Components/Tasks/index.js [hunk 9]
```diff
@@ -1,10 +1,42 @@
...
 function Tasks() {
+  const [tasks, setTask] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [selectedId, setSelectedId] = useState();
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API_URL}/tasks`)
+      .then((response) => response.json())
+      .then((response) => {
+        console.log(response);
```
Comment (lmalvasia) [inline src/Components/Tasks/index.js:15]: remove this console log

**Review by lmalvasia — APPROVED** [review]

File: src/Components/Tasks/Modal/index.jsx (hunk 8 above)
Comment (LucianoFranchino) [reply src/Components/Tasks/Modal/index.jsx:7]: Resolved

File: src/Components/Tasks/index.js (hunk 9 above)
Comment (LucianoFranchino) [reply src/Components/Tasks/index.js:15]: Resolved

**Review by lmalvasia — APPROVED** [review]

**Comment by eyrent** [comment]
### QA Review
- List tasks ✅
- Delete task ✅
- Confirmation modal ✅

**PR APPROVED FOR MERGE**

**Pull request merged**